lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php

```diff
@@ -272,17 +274,18 @@ public function getListViewsSQL(string $database) : string
     }
 
     /**
-     * {@inheritDoc}
+     * @param string $tableWithSchema The name of the table prefixed with the schema (i.e. 'some_schema.some_table').
      */
-    public function getListTableForeignKeysSQL(string $table, ?string $database = null) : string
+    public function getListTableForeignKeysSQL(string $tableWithSchema, ?string $database = null) : string
     {
         return 'SELECT quote_ident(r.conname) as conname, pg_catalog.pg_get_constraintdef(r.oid, true) as condef
                   FROM pg_catalog.pg_constraint r
                   WHERE r.conrelid =
                   (
                       SELECT c.oid
                       FROM pg_catalog.pg_class c, pg_catalog.pg_namespace n
-                      WHERE ' . $this->getTableWhereClause($table) . " AND n.oid = c.relnamespace
+                      WHERE ' . $this->getTableWhereClause($tableWithSchema) . " AND n.oid = c.relnamespace
+                      LIMIT 1
```

Comment:
Instead of failing the query and letting the user know that they are using the API incorrectly, we're hiding the issue and returning one random value instead. I do not agree with this.

Reply:
@greg0ire You approved this PR, so why are you giving this a thumbs up?

We are letting the user know that they are using the API incorrectly thru the deprecation message. That's what we agreed on in a previous conversation.

IMO, this is improper use of SQL. When this crashed for me, the error had no useful meaning as to why. I had to dig deep and debug this code myself to find out the problem was in this SQL - not because I was using the API incorrectly. The API was allowing me to pass the table name without a schema. This fix will provide a useful warning, and in the next version it won't allow it.

If you have a specific way you want this fixed, then either fix it your self, or be clear about exactly how you want it fixed. Stating your opinion isn't helpful if you aren't going to provide a solution. Especially since @greg0ire and I have already had this conversation.

Reply:
@jwilson-cee please do not resolve the conversations that you didn't start.

Reply:
> @greg0ire You approved this PR, so why are you giving this a thumbs up?

Because I mistakenly approved the PR, and I realized @morozov was right, `LIMIT 1` will have the effect of returning something random if you have several schemas (malicious compliance, remember), and shouldn't be needed when using the function properly, right?

> When this crashed for me, the error had no useful meaning as to why.

This is why I proposed this:

> catch the current exception when it happens, and wrap it in an \InvalidArgumentException with a meaningful message if and only if the database schema name is not provided by the user. If somebody can think of a use case for this, they will see the meaningful message, and we will use IN, and add a test that document the use case.

That wouldn't be a BC-break if there already is a crash IMO. Not sure why you didn't address this in your answer (https://github.com/doctrine/dbal/pull/3422#discussion_r340834148). Maybe I wasn't clear enough about this applying to the stable branch.

Reply:
> please do not resolve the conversations that you didn't start.

@morozov Sorry if that annoys you, that was not my intent.

When I've addressed a conversation and I feel that I don't need anything more from it, I'll mark it as resolved. As is encouraged in githubs docs https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request#resolving-conversations

This makes it easier for me, the requestor, to see what conversations still need my attention.

Thanks.